src/qcodes/dataset/data_set_in_memory.py

```diff
@@ -298,8 +298,7 @@ def _load_from_netcdf(
             export_info=export_info,
```

Comment:
Above we should use open_dataset to lazily open the dataset and pass a reference to the dataset path down to the laze cache to avoid loading any real data before its required